### ISSUE TITLE
Don't use native element types directly (label: universal)

### ISSUE THREAD

**Issue by paullryan** [angular/material2#523]
#### Bug, feature request, or proposal:

Bug

#### What is the expected behavior?

material2 components should be compilable by a server side renderer such as angular-universal

#### What is the current behavior?

Compile fails because of direct use of native elements such as HTMLElement.

#### What are the steps to reproduce?

Follow the readme for my test repo at https://github.com/SimplyComplexCo/material-universal-test. Once overlay is functional adding providers for more material2 components will mostly also fail. 

#### What is the use-case or motivation for changing an existing behavior?

Angular Universal support, I understand that #308 is creating a consistent test to keep this functionality in mind.  

This issue is to spur the parallel change of making the current functionality work with Angular Universal.

#### Which versions of Angular, Material, OS, browsers are affected?

Angular Material master

**Comment by paullryan** [comment]
Instead of HTMLElement and ElementRef should be used such as
```ts
foo(element: ElementRef) {
  let nativeElement: any = element.nativeElement;
  // Do thing with native element
}
```

**Comment by jelbourn** [comment]
This is going to be addressed as part of #308
